ci/convert.py

```diff
@@ -0,0 +1,67 @@
+from xml.sax.saxutils import escape
+
+with open('build/res.txt', 'r') as f:
+  lines = f.readlines()
+
+output = '<?xml version="1.0" encoding="UTF-8"?>'
+
+meta = {'duration' : 0}
+suiteName = None
+error = None
+for line in lines:
+  line = line.strip()
+
+  if 'Entering test suite "' in line:
+    start = line.find('Entering test suite "')
+    suiteName = line[start + 21:-2]
+    meta[suiteName] = {'time': 0, 'failures': 0, 'tests' : 0}
+    meta[suiteName]['cases'] = []
+
+  if 'error: in' in line:
+    error = line
+
+  if 'Leaving test case' in line:
+    start = line.find('; testing time: ')
+    tt = 0
+    if start > 0:
+      lineType = line[start + 16:len(line)]
+      if 'mks' in lineType:
+        tt = float(line[start + 16:len(line) - 3]) * 0.001
+      elif 'ms' in lineType:
+        tt = float(line[start + 16:len(line) - 2]) * 0.001
+      else:
+        tt = float(line[start + 16:len(line) - 2]) * 0.000001
```

Comment:
It might be safer to check for `us` here and then just throw an error if it's anything else, but totally up to you.  I don't know if it's worth the effort. :)